packages/rulesets/src/oas/index.ts

```diff
@@ -361,6 +361,18 @@ const ruleset = {
         function: refSiblings,
       },
     },
+    'array-items': {
+      formats: [oas3_0],
+      message: 'Schemas with "type: array", require a sibling "items" field',
+      severity: 0,
+      recommended: true,
+      resolved: false,
+      given: "$..[?(@.type === 'array')]",
+      then: {
+        function: truthy,
```

Comment:
should this detect a child schema?

Reply:
Good question! It should, I added a separate test for nested schemas: https://github.com/stoplightio/spectral/pull/2632/files#diff-7c7ca1323e671563e8696b6c8f4c6b7376a94a6cb93e08dfcec0774957b5a494R57-R63